File: sdkwork-ai-sdk-kotlin/src/main/kotlin/com/sdkwork/ai/KnowledgeSearchRequest.kt
package com.sdkwork.ai

data class KnowledgeSearchRequest(
    val query: String? = null,
    val filters: List<Filter>? = null,
    val knowledgeBaseId: String? = null,
    val topK: Int? = null,
    val scoreThreshold: Double? = null
)

File: sdkwork-ai-sdk-csharp/Models/KnowledgeSearchRequest.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace Ai.Models
{
    public class KnowledgeSearchRequest
    {
        public string? Query { get; set; }
        public List<Filter>? Filters { get; set; }
        public string? KnowledgeBaseId { get; set; }
        public int? TopK { get; set; }
        public double? ScoreThreshold { get; set; }
    }
}
